### PR TITLE
Fixes #22144: Highest id only fetch at first run, but batch is running

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/22144

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Commit modified

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/campaigns/JSONReportsHandler.scala [hunk 1]
```diff
@@ -70,19 +71,33 @@ case class JSONReportsAnalyser(reportsRepository: ReportsRepository, propRepo: R
...
+      _          <- CampaignLogger.trace(s"Got parsed id in ${t2 - t1} ms")
       lowerId     = optLowerId.getOrElse(highestId)
+      _          <- CampaignLogger.debug(s"Last parsed id was ${lowerId}, max id in database is ${highestId}")
       reports    <- reportsRepository.getReportsByKindBetween(lowerId, None, 1000, List(Reports.REPORT_JSON)).toIO
+      t3 <- currentTimeMillis
+      _          <- CampaignLogger.trace(s"Got reports in ${t3 - t2} ms")
 
-      _ <- reports.maxByOption(_._1) match {
-             case None    =>
+      r <- reports.maxByOption(_._1) match {
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/campaigns/JSONReportsHandler.scala:89]: r is unused, should be replaced by _

**Comment by fanf** [comment]
LGTM to me appart formatting/r, can be mered afterward

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
OK, merging this PR

**Pull request merged**